File: ngx-sdc/src/app/core/models/sdc/uri.model.ts
export enum UriType {
  GIT = 'GIT',
  SONAR = 'SONAR'
}

export interface IUriDTO {
  name: string;
  api: string;
  url: string;
}

export interface IUriModel {
  name: string;
  api: string;
  url: string;
}

export namespace IUriModel {
  export const fromDTO = (dto: IUriDTO): IUriModel => new UriModel(dto.name, dto.api, dto.url);
  export const toDTO = (model: IUriModel): IUriDTO => ({ ...model });
}

export class UriModel {
  constructor(
    public name: string,
    public api: string,
    public url: string
  ) {}
}

File: rws-sdc/src/main/java/com/shagui/sdc/json/model/UriModel.java
package com.shagui.sdc.json.model;

import java.util.List;

import com.shagui.sdc.enums.UriType;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class UriModel {
	private String name;
	private String api;
	private String url;
	private UriType type;
	private List<RequestPropertiesModel> properties;
}
